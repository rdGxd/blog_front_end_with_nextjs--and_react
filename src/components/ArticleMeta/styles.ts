import styled, { css } from "styled-components";

export const Wrapper = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.darkerGray};
    font-style: italic;

    .categories span::after {
      content: ", ";
    }

    .categories span:last-child::after {
      content: "";
    }

    a:hover {
      filter: brightness(50%);
    }
  `}
`;
