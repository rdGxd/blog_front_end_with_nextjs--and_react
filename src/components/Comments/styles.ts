import styled, { css } from "styled-components";

export const Wrapper = styled.div`
  ${({ theme }) => css`
    max-width: ${theme.sizes.content};
    width: 100%;
    margin: ${theme.spacings.large} 0 auto;
    padding: 0 ${theme.spacings.large};
  `}
`;
