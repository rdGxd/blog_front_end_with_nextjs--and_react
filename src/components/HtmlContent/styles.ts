import styled, { css } from "styled-components";

export const Container = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};

    p {
      margin: ${theme.spacings.medium} 0;
    }

    a:visited,
    a:link {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 300ms ease-in-out;
    }

    a:hover {
      filter: brightness(80%);
    }

    img {
      max-width: 100%;
    }

    .image {
      background: ${theme.colors.mediumGray};
      line-height: 0;
      margin: ${theme.spacings.medium} 0;
    }

    .image figcaption {
      text-align: center;
      font-size: ${theme.font.sizes.small};
      padding: ${theme.spacings.small};
      line-height: 1.3;
    }

    .image-style-side {
      max-width: 50%;
      float: right;
      margin: ${theme.spacings.medium} 0;
    }

    hr {
      border: none;
      border-bottom: 0.1rem solid ${theme.colors.mediumGray};
    }

    ul,
    ol {
      margin: ${theme.spacings.medium} ${theme.spacings.xlarge};
    }

    .table {
      width: 100%;
      border-collapse: collapse;
      margin: ${theme.spacings.medium} 0;
    }

    table {
      width: 100%;
      border-collapse: collapse;
    }

    table td,
    table th {
      padding: ${theme.spacings.small};
      border: 0.1rem solid ${theme.colors.mediumGray};
    }

    blockquote {
      border-left: 0.5rem solid ${theme.colors.secondary};
      color: ${theme.colors.darkerGray};
      filter: brightness(80%);
      padding-left: ${theme.spacings.medium};
      font-style: italic;
      margin: ${theme.spacings.medium};
    }

    @media ${theme.media.lteMedium} {
      font-size: 2rem;

      .image-style-side {
        max-width: 100%;
        float: none;
        margin: 0;
      }
    }
  `}
`;
