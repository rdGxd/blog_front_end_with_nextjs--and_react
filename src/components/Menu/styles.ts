import styled from "styled-components";

export const Wrapper = styled.div``;

export const Nav = styled.nav``;

export const Logo = styled.nav``;

export const OpenClose = styled.a``;
