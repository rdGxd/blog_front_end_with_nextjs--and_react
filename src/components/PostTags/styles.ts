import styled, { css } from "styled-components";

export const Wrapper = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.medium} 0;

    span {
      margin: 0 0rem 0 0.5rem;
    }

    span::before {
      content: " ";
    }

    span::after {
      content: ", ";
    }

    span:last-child::after {
      content: "";
    }

    a:hover {
      filter: brightness(80%);
    }
  `}
`;
