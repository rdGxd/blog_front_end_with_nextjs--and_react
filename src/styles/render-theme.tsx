import { BlogThemeProvider } from "../contexts/BlogThemeContext";
import { render, RenderResult } from "@testing-library/react";

export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(<BlogThemeProvider>{children}</BlogThemeProvider>);
};
