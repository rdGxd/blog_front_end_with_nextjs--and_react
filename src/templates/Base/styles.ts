import styled, { css } from "styled-components";

export const Wrapper = styled.div``;

export const HeaderContainer = styled.header`
  ${({ theme }) => css`
    ${theme.spacings.xhuge}
  `}
`;

export const ContentContainer = styled.div`
  ${({ theme }) => css`
    max-width: 120rem;
    width: 100%;
    margin: 0 auto;
    margin-bottom: ${theme.spacings.xhuge};
  `}
`;

export const FooterContainer = styled.footer`
  ${({ theme }) => css`
    max-width: 120rem;
    width: 100%;
    margin: 0 auto;
    padding: ${theme.spacings.large};
  `}
`;
