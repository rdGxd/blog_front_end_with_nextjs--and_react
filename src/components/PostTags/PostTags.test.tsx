import { screen } from "@testing-library/react";
import { PostTags } from ".";
import { renderTheme } from "../../styles/render-theme";

import mockPostTags from "./mock";

describe("<PostTags/>", () => {
  it("should render two tags", () => {
    renderTheme(<PostTags {...mockPostTags} />);

    expect(screen.getAllByRole("link")).toHaveLength(2);
    expect(screen.getByText(/Tags:/i)).toBeInTheDocument();
  });

  it("should match snapshot", () => {
    const { container } = renderTheme(<PostTags {...mockPostTags} />);

    expect(container).toMatchSnapshot();
  });

  it("should match snapshot with no tags", () => {
    const { container } = renderTheme(<PostTags {...mockPostTags} tags={undefined} />);

    expect(container).toMatchSnapshot();
  });
});
