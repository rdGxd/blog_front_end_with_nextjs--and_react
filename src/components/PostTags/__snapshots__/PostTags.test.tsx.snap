// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<PostTags /> > should should match snapshot 1`] = `
.c0 {
  margin: 2.4rem auto;
  max-width: 80rem;
  padding: 0 3.2rem 0;
}

.c0 span {
  margin: 0 0rem 0 0.5rem;
}

.c0 span::before {
  content: " ";
}

.c0 span::after {
  content: ", ";
}

.c0 span:last-child::after {
  content: "";
}

.c0 a:hover {
  filter: brightness(80%);
}

<div>
  <p
    class="c0"
  >
    tags:
     
    <span>
      <a
        href="/tag/bc"
      >
        BC
      </a>
    </span>
    <span>
      <a
        href="/tag/bc2"
      >
        BC2
      </a>
    </span>
  </p>
</div>
`;

exports[`<PostTags /> > should should match snapshot with no tags 1`] = `
.c0 {
  margin: 2.4rem auto;
  max-width: 80rem;
  padding: 0 3.2rem 0;
}

.c0 span {
  margin: 0 0rem 0 0.5rem;
}

.c0 span::before {
  content: " ";
}

.c0 span::after {
  content: ", ";
}

.c0 span:last-child::after {
  content: "";
}

.c0 a:hover {
  filter: brightness(80%);
}

<div>
  <p
    class="c0"
  >
    tags:
     
  </p>
</div>
`;
